server/src/main/java/org/elasticsearch/common/blobstore/BlobStore.java

```diff
@@ -27,8 +29,25 @@ public interface BlobStore extends Closeable {
      * Delete all the provided blobs from the blob store. Each blob could belong to a different {@code BlobContainer}
      * @param blobNames the blobs to be deleted
      */
+    @Deprecated(forRemoval = true)
     void deleteBlobsIgnoringIfNotExists(Iterator<String> blobNames) throws IOException;
 
+    // TODO: Remove the default implementation and require each blob store to implement this method. Once it's done, remove the
+    // the above overload version that does not take the Purpose parameter.
+    /**
+     * Delete all the provided blobs from the blob store. Each blob could belong to a different {@code BlobContainer}
+     * @param blobNames the blobs to be deleted
+     * @param purpose the purpose of the {@code BlobContainer} associated to the blobs to be deleted. It should be set
+     *                to {@code Purpose.GENERIC}, if the blobs are from multiple {@code BlobContainer}s.
+     */
+    default void deleteBlobsIgnoringIfNotExists(Iterator<String> blobNames, Purpose purpose) throws IOException {
+        if (purpose == Purpose.GENERIC) {
+            deleteBlobsIgnoringIfNotExists(blobNames);
+        } else {
+            throw new UnsupportedOperationException();
+        }
+    }
```

Comment:
As David [suggested](https://github.com/elastic/elasticsearch/pull/99615#discussion_r1328331423), I will perform the cleanup of these two methods in a follow-up and avoid breaking builds on either side.